infrastructure/010-aws-api/modules/dynamodb/dynamodb.tf

```diff
@@ -0,0 +1,53 @@
+module "this" {
+  source = "terraform-aws-modules/dynamodb-table/aws"
+
+  name      = "${var.name_prefix}-${random_id.this.hex}"
+  hash_key  = "PartitionKey"
+  range_key = "SortKey"
+
+  # Enable streaming to allow for global replication
+  stream_enabled   = var.is_global
+  stream_view_type = var.is_global ? "NEW_AND_OLD_IMAGES" : null
+
+  server_side_encryption_enabled     = true
+  server_side_encryption_kms_key_arn = var.server_side_encryption_kms_key_arn
+
+  attributes = [
+    {
+      name = "PartitionKey"
+      type = "S"
+    },
+    {
+      name = "SortKey"
+      type = "S"
+    },
+    {
+      name = "Value"
+      type = "S"
+    }
+  ]
+
+  global_secondary_indexes = [
+    {
+      hash_key        = "SortKey"
+      name            = "InverseIndex"
+      projection_type = "KEYS_ONLY"
+      range_key       = "PartitionKey"
+    },
```

Comment:
Creating an index that allows reverse lookup of partition vs sort key allows for selection of collection resources and inverted relations. We use KEYS_ONLY because this should only be used to lookup the next partition key and that should be used for authority on reads to prevent lags. Since the sort key may have an uneven keyspace this also helps prevent hot spots.